wallets/src/main/java/network/misq/wallets/bitcoind/BitcoindProcess.java

```diff
@@ -0,0 +1,98 @@
+package network.misq.wallets.bitcoind;
+
+import network.misq.wallets.NetworkType;
+import network.misq.wallets.rpc.config.RpcConfig;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+
+public class BitcoindProcess {
+
+    private final RpcConfig rpcConfig;
+    private final Path dataDir;
+
+    private Process bitcoindProcess;
+
+    public BitcoindProcess(RpcConfig rpcConfig, Path dataDir) {
+        this.rpcConfig = rpcConfig;
+        this.dataDir = dataDir;
+    }
+
+    public void startAndWaitUntilReady() {
+        dataDir.toFile().mkdirs();
+        bitcoindProcess = createAndStartProcess();
+        waitUntilReady();
+    }
+
+    public void stopAndWaitUntilStopped() {
+        int exitCode = 1;
+        try {
+            while (exitCode != 0) {
+                Process p = new ProcessBuilder(
+                        "bitcoin-cli",
+                        rpcConfig.networkType() == NetworkType.REGTEST ? "-regtest" : "",
+                        "-rpcuser=" + rpcConfig.user(),
+                        "-rpcpassword=" + rpcConfig.password(),
+                        "stop"
+                ).start();
+                exitCode = p.waitFor();
+
+                bitcoindProcess.waitFor();
```

Comment:
When running the tests it takes quite a while until the process is terminated.  Not sure what happens if bitcoind gets shut down too quickly, probably for mainnet not good, but for tests with regtest might be ok to speed up test runs. 

We could pass a timeout and use the matching method signature.  Maybe have one `stopAndWaitUntilStopped()` signature with timeout and one without?

`if (!bitcoindProcess.waitFor(100, TimeUnit.MILLISECONDS)) {
           log.warn("Terminated bitcoindProcess by a timeout");
 }`

Reply:
I added this but if I use it in the tests they never complete. Starting a new regtest instance before the previous has terminated seems to be a bad idea. We could use different ports but this would start a lot of parallel instances. Do we want this?

Reply:
No, I think 1 single instance of bitcoind is better. But maybe it was a bad suggestion anyway. I was just wondering why the shutdown takes so long. I just tested with normal Bitcoin core regtest and at quit it shuts down very fast. So might be some other issue. Also the test I could run seemed to be quite slow. So maybe there is some blocking issues somewhere which also causes my issues running the tests. How long does it take for you to run for instance BitcoindWalletCreationTests? For me after the test is green it takes about 10 sec until the test runner exits.

Reply:
BitcoindWalletCreationTests runs for 8 seconds but the shutdown takes the most time. Yes, I'm not sure why it takes so long. Hmm I'll try to find out why it takes so long.